src/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -479,19 +480,26 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             CancellationTokenSource cts;
             bool disposeCts;
             bool hasTimeout = _timeout != s_infiniteTimeout;
+            CancellationTokenSource pendingRequestsCts = _pendingRequestsCts;
+            Func<Exception, bool> isTimeout = NoTimeoutPredicate;
             if (hasTimeout || cancellationToken.CanBeCanceled)
             {
                 disposeCts = true;
-                cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _pendingRequestsCts.Token);
+                cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, pendingRequestsCts.Token);
                 if (hasTimeout)
                 {
+                    isTimeout = e =>
+                        e is OperationCanceledException
+                        && cts.IsCancellationRequested
+                        && !pendingRequestsCts.IsCancellationRequested
```

Comment:
I see that you're distinguishing between cancellations triggered by timeout and ambient cancellation tokens. While it might be the right choice, it's a design decision that needs to be stated explicitly and discussed.

Reply:
I didn't have the impression of consciously making a design decision here. I mean, if the request is canceled by a call to `CancelPendingRequests()`, it's not a timeout, is it?
Do you have a different perspective on this?

Reply:
I don't really.